File: R/example8_2.R
# Example 8.2 Logit Model of Economic Education
# Binomial Regression Model: Probit and Logit

setwd("C:/Course17/ceR/R")
data<-read.table("http://web.pdx.edu/~crkl/ceR/data/grade.txt",header=T,nrows=32)

gpa<-data$GPA
tuce<-data$TUCE
psi<-data$PSI
grade<-data$GRADE

model1<-glm(grade~gpa+tuce+psi,binomial)  # default link=logit
summary(model1)

# logit model interpretation
b1<-model1$coefficients
# estimated probability
probility<-predict(model1,type="response")       

# marginal effects of an explanatory variable
gpa_slopes<-probility*(1-probility)*b1["gpa"]
tuce_slopes<-probility*(1-probility)*b1["tuce"]
psi_slopes<-probility*(1-probility)*b1["psi"]

cbind(grade,probility,gpa_slopes,tuce_slopes,psi_slopes)

rm(list=ls())

File: python/example8_2.py
# Example 8.2 Logit Model of Economic Education
# Binomial Regression Model: Probit and Logit
# Logit Model using statsmodels package

import numpy as np
import pandas as pd
import statsmodels.api as sm
import statsmodels.formula.api as smf 

data = pd.read_csv('http://web.pdx.edu/~crkl/ceR/data/grade.txt',sep='\s+', nrows=32) 
data = data.dropna()
print(data.describe())

X = data[['GPA','TUCE','PSI']]
y = data['GRADE']
model1 = sm.Logit(y,sm.add_constant(X)).fit()
model1.summary()

model2 = sm.Logit.from_formula('GRADE ~ GPA + TUCE + PSI',data=data).fit()
model2.summary()

# same as model2
model3 = smf.logit('GRADE ~ GPA + TUCE + PSI', data=data).fit()
model3.summary()

# using GLM
model4 = smf.glm(formula='GRADE ~ GPA + TUCE + PSI', 
                family=sm.families.Binomial(),
                data=data).fit()
model4.summary()

# probit model interpretation
model2.params
# predicted probability of each obs
model2.predict(data)
# marginal effects at the mean of each regressor
margeff = model2.get_margeff('mean')
margeff.summary()
